modules/engine/src/shader-inputs.ts

```diff
@@ -89,9 +89,10 @@ export class ShaderInputs<
       const moduleProps = props[moduleName];
       const module = this.modules[moduleName];
 
-      const uniforms = module.getUniforms?.(moduleProps, this.moduleUniforms[moduleName]);
+      const oldUniforms = this.moduleUniforms[moduleName] || {};  
+      const uniforms = module.getUniforms?.(moduleProps, this.moduleUniforms[moduleName]) || moduleProps as any;
       // console.error(uniforms)
-      this.moduleUniforms[moduleName] = uniforms || moduleProps as any;
+      this.moduleUniforms[moduleName] = {...oldUniforms, ...uniforms};
```

Comment:
There's no nesting within a given module's uniforms, correct?

Reply:
Correct, the type is `PropsT extends Record<string, unknown> `